habitat_ros/hab_ros_interface.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import rospy
+from rospy.numpy_msg import numpy_msg
+from rospy_tutorials.msg import Floats
+from geometry_msgs.msg import Twist
+import threading
+import sys
+
+sys.path = [
+    b for b in sys.path if "2.7" not in b
+]  # hacky solution, removes ROS related paths or else certain packages like cv2 can't be imported
+
+import habitat
+import habitat_sim.bindings as hsim
+import magnum as mn
+import numpy as np
+import time
+import cv2
+
+count = 0
+
+lock = threading.Lock()
+rospy.init_node("plant_model", anonymous=True)
+
+
+class sim_env(threading.Thread):
+
+    _x_axis = 0
+    _y_axis = 1
+    _z_axis = 2
+    _dt = 0.00478
+    _sensor_rate = 80  # hz
+    _r = rospy.Rate(_sensor_rate)
+
+    def __init__(self, env_config_file):
+        threading.Thread.__init__(self)
+        self.env = habitat.Env(config=habitat.get_config(env_config_file))
+        self.env._sim._sim.agents[0].move_filter_fn = self.env._sim._sim._step_filter
+        self.observations = self.env.reset()
+
+        self.env._sim._sim.agents[0].state.velocity = np.float32([0, 0, 0])
+        self.env._sim._sim.agents[0].state.angular_velocity = np.float32([0, 0, 0])
+
+        self._pub_rgb = rospy.Publisher("rgb", numpy_msg(Floats), queue_size=1)
+        self._pub_depth = rospy.Publisher("depth", numpy_msg(Floats), queue_size=1)
+        self._pub_pose = rospy.Publisher("agent_pose", numpy_msg(Floats), queue_size=1)
+        self._pub_depth_and_pointgoal = rospy.Publisher(
+            "depth_and_pointgoal", numpy_msg(Floats), queue_size=1
+        )
+        print("created habitat_plant succsefully")
+
+    def _render(self):
+        self.env._update_step_stats()  # think this increments episode count
+        sim_obs = self.env._sim._sim.get_sensor_observations()
+        self.observations = self.env._sim._sensor_suite.get_observations(sim_obs)
+        self.observations.update(
+            self.env._task.sensor_suite.get_observations(
+                observations=self.observations, episode=self.env.current_episode
+            )
+        )
+
+    def _update_position(self):
+        state = self.env.sim.get_agent_state(0)
+        vz = -state.velocity[0]
+        vx = state.velocity[1]
+        dt = self._dt
+
```

Comment:
```
scene_node = self.env._sim._sim.agents[0].scene_node
```
Then all the `self.env._sim._sim.agents[0].scene_node` bellow can be replaced with `scene_node`.

Reply:
Actions with state (like these) is something we should support.  Then we'd just be able to add a new action for the agent to take (for both variants).